hpx/runtime/threads/policies/thread_queue.hpp

```diff
@@ -215,8 +213,6 @@ namespace hpx { namespace threads { namespace policies
 
             else
             {
-                hpx::util::unlock_guard<Lock> ull(lk);
-
```

Comment:
AFAIR, this unlock was added to avoid a potentially held lock while suspension problem. Are you sure this doesn't happen anymore?